glusterfs/datadog_checks/glusterfs/check.py

```diff
@@ -0,0 +1,207 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import json
+import os
+from typing import Any
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+from datadog_checks.base.config import _is_affirmative
+from datadog_checks.base.utils.subprocess_output import get_subprocess_output
+
+CLUSTER_STATS = {
+    'node_count': 'nodes.count',
+    'nodes_active': 'nodes.active',
+    'volumes_started': 'volumes.started',
+    'volume_count': 'volumes.count',
+}
+
+
+GENERAL_STATS = {
+    'size_used': 'size.used',
+    'inodes_free': 'inodes.free',
+    'size_free': 'size.free',
+    'size_total': 'size.total',
+    'inodes_used': 'inodes.used',
+    'inodes_total': 'inodes.total',
+    'online': 'online',
+}
+
+VOL_SUBVOL_STATS = {
+    'disperse': 'disperse',
+    'disperse_redundancy': 'disperse_redundancy',
+    'replica': 'replica',
+}
+
+VOLUME_STATS = {
+    'v_used_percent': 'used.percent',
+    'num_bricks': 'bricks.count',
+    'distribute': 'distribute',
+    'v_size_used': 'size.used',
+    'v_size': 'size.total',
+    'snapshot_count': 'snapshot.count',
+}
+
+VOLUME_STATS.update(GENERAL_STATS)
+VOLUME_STATS.update(VOL_SUBVOL_STATS)
+
+BRICK_STATS = {'block_size': 'block_size'}
+BRICK_STATS.update(GENERAL_STATS)
+
+PARSE_METRICS = ['v_size_used', 'v_size']
+
+GLUSTER_VERSION = 'glfs_version'
+CLUSTER_STATUS = 'cluster_status'
+
+GSTATUS_PATH = '/opt/datadog-agent/embedded/sbin/gstatus'
+INSTALL_PATH = '/usr/local/bin/gstatus'
+
+
+class GlusterfsCheck(AgentCheck):
+    __NAMESPACE__ = 'glusterfs'
+
+    CLUSTER_SC = "cluster.health"
+    VOLUME_SC = "volume.health"
+    BRICK_SC = "brick.health"
+
+    def __init__(self, name, init_config, instances):
+        # type: (*Any, **Any) -> None
+        super(GlusterfsCheck, self).__init__(name, init_config, instances)
+        self._tags = self.instance.get('tags', [])
+
+        # Check if customer set gstatus path
+        if init_config.get('gstatus_path'):
+            self.gstatus_cmd = init_config.get('gstatus_path')
+        else:
+            if os.path.exists(GSTATUS_PATH):
+                self.gstatus_cmd = GSTATUS_PATH
+            elif os.path.exists(INSTALL_PATH):
+                self.gstatus_cmd = INSTALL_PATH
+            else:
+                raise ConfigurationError(
+                    'Glusterfs check requires `gstatus` to be installed or set the path to the installed version.'
+                )
+        self.log.debug("Using gstatus path `%s`", self.gstatus_cmd)
+
+    def check(self, _):
+        use_sudo = _is_affirmative(self.instance.get('use_sudo', False))
+        if use_sudo:
+            test_sudo = os.system('setsid sudo -l < /dev/null')
```

Comment:
Your running `sudo -l` as a daemon. I assume that `setsid` returns code will always be 0, right? No matter what `sudo -l` returns.

Also `sudo -l` is tricky as it might ask for a password.
Last, why do we use `os.system` instead of `get_subprocess_output`?

Maybe something like `test_sudo = get_subprocess_output(['sudo', '-ln', self.gstatus_cmd])` and then checking the return code of that call?

Reply:
Updated!